heron/stmgr/src/cpp/manager/stmgr.cpp

```diff
@@ -746,5 +746,15 @@ void StMgr::SendStartBackPressureToOtherStMgrs() {
 
 void StMgr::SendStopBackPressureToOtherStMgrs() { clientmgr_->SendStopBackPressureToOtherStMgrs(); }
 
+// Do any actions if a stmgr client connection dies
+void StMgr::HandleDeadStMgrConnection(const sp_string&) {
+  // TODO(srkukarni) For Stateful Topologies, we need to do a restore
```

Comment:
This will become clearer in the following prs.
But essentially the intuition is that when a stmgr notices that its connection to another stmgr is gone, it should initiate certain actions. Like for instance, if it was running normally, it should ask the tmaster to start a recover topololgy from last sane checkpoint.